### PR TITLE
Updated citation information.

### PR DESCRIPTION
**Pull request by bdice**
## Description
Updates citation information to point to the new paper in Computer Physics Communications.

## Types of changes

- [x] Docs

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/glotzerlab/freud/blob/master/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [x] I have updated the documentation (if relevant).

### CODE REVIEW

**Comment by vyasr** [comment]
@bdice I think it's safe to update this now, if it's in Draft because you were waiting for official publication I think we're close enough now that it's safe to make the change.

**Comment by bdice** [comment]
@vyasr Does this PR need to be updated somehow with your recent work into software citations?

**Comment by vyasr** [comment]
No, that's completely independent of this and just needs to be updated in parallel. I received a communication that the paper is scheduled to be published in the upcoming issue of CPC, so we can either merge this as is and then update again, or wait for that so that we don't have to update twice. Up to you, I'm fine with getting this version on our webpage sooner even if it's still officially in press.

File: doc/source/reference/citing.rst [hunk 1]
```diff
@@ -6,21 +6,28 @@ Please acknowledge the use of this software within the body of your publication
 
 *Data analysis for this publication utilized the freud library[1].*
 
-  [1] V. Ramasubramani, B. D. Dice, E. S. Harper, M. P. Spellings, J. A. Anderson, and S. C. Glotzer. freud: A Software Suite for High Throughput Analysis of Particle Simulation Data. 2019. arXiv:1906.06317.
+  [1] V. Ramasubramani, B. D. Dice, E. S. Harper, M. P. Spellings, J. A. Anderson, and S. C. Glotzer. freud: A Software Suite for High Throughput Analysis of Particle Simulation Data. Computer Physics Communications 2020. doi:10.1016/j.cpc.2020.107275.
```
Comment (bdice) [inline doc/source/reference/citing.rst]:
```suggestion
  [1] V. Ramasubramani, B. D. Dice, E. S. Harper, M. P. Spellings, J. A. Anderson, and S. C. Glotzer. freud: A Software Suite for High Throughput Analysis of Particle Simulation Data. Computer Physics Communications Volume 254, September 2020, 107275. doi:10.1016/j.cpc.2020.107275.
```

File: doc/source/reference/citing.rst [hunk 2]
```diff
@@ -6,21 +6,28 @@ Please acknowledge the use of this software within the body of your publication
...
                   Eric S. Harper and
                   Matthew P. Spellings and
                   Joshua A. Anderson and
                   Sharon C. Glotzer},
-        title = {freud: A Software Suite for High Throughput
-                 Analysis of Particle Simulation Data},
-        year = {2019},
-        eprint = {arXiv:1906.06317},
+        journal = {Computer Physics Communications},
+        pages = {107275},
```
Comment (bdice) [inline doc/source/reference/citing.rst:28]:
```suggestion
        volume = {254},
        pages = {107275},
```

File: freud/__init__.py [hunk 3]
```diff
@@ -45,15 +45,21 @@
...
               Eric S. Harper and
               Matthew P. Spellings and
               Joshua A. Anderson and
               Sharon C. Glotzer},
-    title = {freud: A Software Suite for High Throughput
-             Analysis of Particle Simulation Data},
-    year = {2019},
-    eprint = {arXiv:1906.06317},
+    journal = {Computer Physics Communications},
+    pages = {107275},
```
Comment (bdice) [inline freud/__init__.py:59]:
```suggestion
    volume = {254},
    pages = {107275},
```

**Review by vyasr — APPROVED** [review]

**Pull request merged**